src/arithmetic/algebraic_expression/algebraic_expression_eval.c

```diff
@@ -135,21 +137,21 @@ static GrB_Matrix _Eval_Mul(const AlgebraicExpression *exp, GrB_Matrix res) {
 	GrB_Matrix B;
 	GrB_Info info;
 	GrB_Index nvals;
-	GrB_Descriptor desc;
+	GrB_Descriptor desc = GrB_NULL;
 	AlgebraicExpression *left = CHILD_AT(exp, 0);
 	AlgebraicExpression *right = CHILD_AT(exp, 1);
 
-	GrB_Descriptor_new(&desc);  // Descriptor used for transposing operands.
-
 	if(left->type == AL_OPERATION) {
 		assert(left->operation.op == AL_EXP_TRANSPOSE);
+		if(desc == GrB_NULL) GrB_Descriptor_new(&desc);
 		GrB_Descriptor_set(desc, GrB_INP0, GrB_TRAN);
 		left = CHILD_AT(left, 0);
 	}
 	A = left->operand.matrix;
 
 	if(right->type == AL_OPERATION) {
 		assert(right->operation.op == AL_EXP_TRANSPOSE);
+		if(desc == GrB_NULL) GrB_Descriptor_new(&desc);
```

Comment:
A bit further down, if `B` is the identity matrix, and there's a transpose over it, we should reset the descriptor for `GrB_INP1`